File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_for 'any_channel'
  end

  def unsubscribed
  end

  def post(data)
    pp data
    ChatChannel.broadcast_to('any_channel', name: data['name'], line: data['line'])
  end
end

File: app/javascript/packs/channels/chat_channel.js
import { createConsumer } from '@rails/actioncable'

class ChatChannel {
  constructor(listener, userName) {
    this.listener = listener
    this.userName = userName

    const consumer = createConsumer()
    this.subscription = consumer.subscriptions.create(
      { channel: 'ChatChannel' },
      { received: (data) => this.received(data) }
    )
  }

  post(line) {
    this.subscription.perform('post', { name: this.userName, line: line })
  }

  received(data) {
    this.listener.onPost(data)
  }
}

export default ChatChannel
